controllers/suite_test.go

```diff
@@ -72,36 +70,15 @@ var _ = BeforeSuite(func(done Done) {
 	Expect(err).ToNot(HaveOccurred())
 	Expect(cfg).ToNot(BeNil())
 
-	err = syncv1alpha1.AddToScheme(scheme.Scheme)
-	err = argov1alpha1.AddToScheme(scheme.Scheme)
-	Expect(err).NotTo(HaveOccurred())
+	Expect(syncv1alpha1.AddToScheme(scheme.Scheme)).To(Succeed())
+	Expect(argov1alpha1.AddToScheme(scheme.Scheme)).To(Succeed())
 
 	// +kubebuilder:scaffold:scheme
 
 	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
 	Expect(err).NotTo(HaveOccurred())
 	Expect(k8sClient).NotTo(BeNil())
 
-	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
-		Scheme: scheme.Scheme,
-	})
-	Expect(err).ToNot(HaveOccurred())
-
-	mockAcdClient := mocks.ArgoCDAppClientStub{}
-	reconciler = &ProgressiveSyncReconciler{
-		Client:          k8sManager.GetClient(),
-		Log:             ctrl.Log.WithName("controllers").WithName("progressiverollout"),
-		ArgoCDAppClient: &mockAcdClient,
-	}
-	err = reconciler.SetupWithManager(k8sManager)
```

Comment:
Now that each test in progressive_sync_controller_test.go is going to have its own reconciler, should we add an integration or unit test to prevent regressions on the logic of naming the keys for our in-memory storage #116?

I am saying this because as part of #116, we have added keys for storage that are prefixed by progressive sync name to prevent different tests (different PS in production) from overwriting our storage, with this change to creating a PR once per test instead of per test suite we no longer have a code path that tests that.

Reply:
Shouldn't we tackle that in #116 ? This PR doesn't know anything about that structure, no?